src/lib/settings/settings_reader/recipients_settings_reader.py

```diff
@@ -36,11 +36,11 @@ def get_monitoring_group_names(self) -> list[str]:
         Returns:
             list: List of Monitoring Group names.
         """
-        monitoring_group_names = []
-
-        for recipient in self._recipients:
-            for subscription in recipient.get("subscriptions"):
-                monitoring_group_names.append(subscription.get("monitoring_group"))
+        monitoring_group_names = [
+            subscription.get("monitoring_group")
+            for recipient in self._recipients
+            for subscription in recipient.get("subscriptions")
+        ]
```

Comment:
let's keep method as it is for now (this PR).
Later we need to review some methods.
For example if this is for getting recipient's subscriptions to send alert or digest - than it requires additional parameter ( subscription_type : alerts / digest ) and filter only when subscription_type is true in config

Reply:
yes, currently this method is specially for the validation of all the recepients config.
Probably we would need to make it more generic to provide also the functionality you describing
P.S. For the specific type  currently there is alother method used - get_recipients_by_monitoring_groups_and_type